src/libcore/ffi.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation of a `va_list`.
+
+#[cfg(any(all(not(target_arch = "aarch64"), not(target_arch = "powerpc"),
+              not(target_arch = "x86_64")),
+          windows))]
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+/// Basic implementation of a `va_list`.
+pub struct VaListTy<'a> {
```

Comment:
This is closer to the `va_list` type that is implemented in the example in the [LLVM Language Reference Variable Argument Handling Section](https://llvm.org/docs/LangRef.html#int-varargs)

Reply:
Why is this `pub`?

Reply:
Wait, why can't you just replace this struct with `&'a mut VaListImpl` where:
```rust
#[repr(transparent)]
struct VaListImpl {
    _opaque: u8,
}
```
That should be 100% equivalent, and perhaps it can even be made less weird.
But overall, I think it would get rid of the `VaListTy` complication.

Reply:
Yeah, no need to be `pub`, and I think that would work. I changed to this type so that I was essentially copying what the LLVM Language Reference had to ensure I had `i686` working. Not that I have a working version, I'll try to experiment with something like the above.

Side note: I think this should eliminate some of the hacky `cfg`s in the `VaList::copy` implementation.